diff -Naur lighttpd-1.4.49/src/fdevent.c lighttpd-1.4.49-2/src/fdevent.c
--- lighttpd-1.4.49/src/fdevent.c	2018-03-12 10:52:20.000000000 +1000
+++ lighttpd-1.4.49-2/src/fdevent.c	2018-04-12 16:36:01.504086685 +1000
@@ -561,7 +561,7 @@
 	int fd;
 	socklen_t len = (socklen_t) *addrlen;
 
-      #if defined(SOCK_CLOEXEC) && defined(SOCK_NONBLOCK)
+      #if defined(SOCK_CLOEXEC) && defined(SOCK_NONBLOCK) && __ANDROID_API__ >= 21
        #if defined(__NetBSD__)
 	fd = paccept(listenfd, addr, &len, NULL, SOCK_CLOEXEC | SOCK_NONBLOCK);
        #else
@@ -571,6 +571,12 @@
        #endif
       #else
 	fd = accept(listenfd, addr, &len);
+	if (fd >= 0) {
+		if (0 != fdevent_fcntl_set_nb_cloexec(NULL, fd)) {
+			close(fd);
+			fd = -1;
+		}
+	}
       #endif
 
 	if (fd >= 0) *addrlen = (size_t)len;
