diff -Naur php-5.6.24/ext/standard/basic_functions.c php-5.6.24-2/ext/standard/basic_functions.c
--- php-5.6.24/ext/standard/basic_functions.c	2016-08-31 13:56:02.921792680 +1000
+++ php-5.6.24-2/ext/standard/basic_functions.c	2016-08-31 14:01:02.734647286 +1000
@@ -1836,6 +1836,7 @@
 ZEND_END_ARG_INFO()
 /* }}} */
 /* {{{ password.c */
+#if HAVE_CRYPT
 ZEND_BEGIN_ARG_INFO_EX(arginfo_password_hash, 0, 0, 2)
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, algo)
@@ -1853,6 +1854,7 @@
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, hash)
 ZEND_END_ARG_INFO()
+#endif
 /* }}} */
 /* {{{ proc_open.c */
 #ifdef PHP_CAN_SUPPORT_PROC_OPEN
@@ -2862,13 +2864,15 @@
 	PHP_FE(base64_decode,													arginfo_base64_decode)
 	PHP_FE(base64_encode,													arginfo_base64_encode)
 
+#if HAVE_CRYPT
 	PHP_FE(password_hash,													arginfo_password_hash)
 	PHP_FE(password_get_info,												arginfo_password_get_info)
 	PHP_FE(password_needs_rehash,											arginfo_password_needs_rehash)
 	PHP_FE(password_verify,													arginfo_password_verify)
+#endif
+
 	PHP_FE(convert_uuencode,												arginfo_convert_uuencode)
 	PHP_FE(convert_uudecode,												arginfo_convert_uudecode)
-
 	PHP_FE(abs,																arginfo_abs)
 	PHP_FE(ceil,															arginfo_ceil)
 	PHP_FE(floor,															arginfo_floor)
@@ -3622,7 +3626,9 @@
 	BASIC_MINIT_SUBMODULE(browscap)
 	BASIC_MINIT_SUBMODULE(standard_filters)
 	BASIC_MINIT_SUBMODULE(user_filters)
+#if HAVE_CRYPT
 	BASIC_MINIT_SUBMODULE(password)
+#endif
 
 #if defined(HAVE_LOCALECONV) && defined(ZTS)
 	BASIC_MINIT_SUBMODULE(localeconv)
