diff -Naur php-7.1.14/ext/standard/basic_functions.c php-7.1.14-2/ext/standard/basic_functions.c
--- php-7.1.14/ext/standard/basic_functions.c	2018-01-31 05:05:32.000000000 +1000
+++ php-7.1.14-2/ext/standard/basic_functions.c	2018-02-14 13:56:01.912744693 +1000
@@ -1835,6 +1835,7 @@
 ZEND_END_ARG_INFO()
 /* }}} */
 /* {{{ password.c */
+#if HAVE_CRYPT
 ZEND_BEGIN_ARG_INFO_EX(arginfo_password_hash, 0, 0, 2)
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, algo)
@@ -1852,6 +1853,7 @@
 	ZEND_ARG_INFO(0, password)
 	ZEND_ARG_INFO(0, hash)
 ZEND_END_ARG_INFO()
+#endif
 /* }}} */
 /* {{{ proc_open.c */
 #ifdef PHP_CAN_SUPPORT_PROC_OPEN
@@ -2889,13 +2891,15 @@
 	PHP_FE(base64_decode,													arginfo_base64_decode)
 	PHP_FE(base64_encode,													arginfo_base64_encode)
 
+#if HAVE_CRYPT
 	PHP_FE(password_hash,													arginfo_password_hash)
 	PHP_FE(password_get_info,												arginfo_password_get_info)
 	PHP_FE(password_needs_rehash,											arginfo_password_needs_rehash)
 	PHP_FE(password_verify,													arginfo_password_verify)
+#endif
+
 	PHP_FE(convert_uuencode,												arginfo_convert_uuencode)
 	PHP_FE(convert_uudecode,												arginfo_convert_uudecode)
-
 	PHP_FE(abs,																arginfo_abs)
 	PHP_FE(ceil,															arginfo_ceil)
 	PHP_FE(floor,															arginfo_floor)
@@ -3668,7 +3672,9 @@
 	BASIC_MINIT_SUBMODULE(browscap)
 	BASIC_MINIT_SUBMODULE(standard_filters)
 	BASIC_MINIT_SUBMODULE(user_filters)
+#if HAVE_CRYPT
 	BASIC_MINIT_SUBMODULE(password)
+#endif
 	BASIC_MINIT_SUBMODULE(mt_rand)
 
 #if defined(HAVE_LOCALECONV) && defined(ZTS)

